File: string/0443_string_compression/0443_string_compression.go
package main

import (
	"fmt"
	"strconv"
)

func compress(chars []byte) int {
	end, count := 0, 1

	for i, char := range chars {
		if i+1 < len(chars) && char == chars[i+1] {
			count++
		} else {
			chars[end] = char
			end++

			if count > 1 {
				for _, num := range strconv.Itoa(count) {
					chars[end] = byte(num)
					end++
				}
				count = 1
			}
		}
	}

	return end
}

func main() {
	str := "aabbccc"
	chars := []byte(str)
	res := compress(chars)
	fmt.Println(res)
}

File: string/0443_string_compression/0443_string_compression.py

class Solution(object):
    def compress(self, chars):
        anchor, write = 0, 0
        for read, c in enumerate(chars):
            if read + 1 == len(chars) or chars[read + 1] != c:
                chars[write] = chars[anchor]
                write += 1
                if read > anchor:
                    for digit in str(read - anchor + 1):
                        chars[write] = digit
                        write += 1
                anchor = read + 1
        return write


chars = ["a","a","b","b","c","c","c"]
res = Solution().compress(chars)
print(res)